packages/captp/src/atomics.js

```diff
@@ -0,0 +1,149 @@
+// @ts-check
+/* global BigUint64Array */
+
+import { assert, details as X } from '@agoric/assert';
+
+// This is a pathological minimum, but exercised by the unit test.
+// transfer.
+export const MIN_DATA_BUFFER_LENGTH = 1;
+
+// Calculate how big the transfer buffer needs to be.
+export const TRANSFER_OVERHEAD_LENGTH =
+  BigUint64Array.BYTES_PER_ELEMENT + Int32Array.BYTES_PER_ELEMENT;
+export const MIN_TRANSFER_BUFFER_LENGTH =
+  MIN_DATA_BUFFER_LENGTH + TRANSFER_OVERHEAD_LENGTH;
+
+// These are bit flags for the status element of the transfer buffer.
+const STATUS_WAITING = 1;
+const STATUS_FLAG_DONE = 2;
+const STATUS_FLAG_REJECT = 4;
+
+/**
+ * Return a status buffer, length buffer, and data buffer backed by transferBuffer.
+ *
+ * @param {SharedArrayBuffer} transferBuffer the backing buffer
+ */
+const splitTransferBuffer = transferBuffer => {
+  assert(
+    transferBuffer.byteLength >= MIN_TRANSFER_BUFFER_LENGTH,
+    X`Transfer buffer of ${transferBuffer.byteLength} bytes is smaller than MIN_TRANSFER_BUFFER_LENGTH ${MIN_TRANSFER_BUFFER_LENGTH}`,
+  );
+  const lenbuf = new BigUint64Array(transferBuffer, 0, 1);
+  const statusbuf = new Int32Array(transferBuffer, lenbuf.byteLength, 1);
+  const overheadLength = lenbuf.byteLength + statusbuf.byteLength;
+  assert.equal(
+    overheadLength,
+    TRANSFER_OVERHEAD_LENGTH,
+    X`Internal error; actual overhead ${overheadLength} of bytes is not TRANSFER_OVERHEAD_LENGTH ${TRANSFER_OVERHEAD_LENGTH}`,
+  );
+  const databuf = new Uint8Array(
+    transferBuffer,
+    lenbuf.byteLength + statusbuf.byteLength,
```

Comment:
You already added them together into `overheadLength` and asserted that the sum is equal to `TRANSFER_OVERHEAD_LENGTH`.
```suggestion
    TRANSFER_OVERHEAD_LENGTH,
```